### DISCUSSION TITLE
RuntimeInformation.FrameworkDescription returning back surprising results.

### DISCUSSION THREAD

**Discussion by normj** [dotnet/runtime#98529] [Q&A] [answered]
Our code adds the value of `RuntimeInformation.FrameworkDescription` to our API calls for logging purposes. Most all of the logs look as you would expect with values like `.NET 8.0.2`. In a small percent of our logs we are seeing values like `EXPLICIT/ACB7CD69` show up. Is there a reason `RuntimeInformation.FrameworkDescription` would return back something besides the documented patterns?

> The property returns a string that indicates the name of the currently executing .NET implementation. The string starts with one of the following descriptions, usually followed by the current version:
>
> ".NET" (for .NET 5 and later versions)
> ".NET Core" (for .NET Core 1.0 - 3.1)
> ".NET Framework"
> ".NET Native"
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.frameworkdescription?view=net-8.0

**Comment by elgonzo** [comment]
I don't think values like `EXPLICIT/ACB7CD69` is coming from `RuntimeInformation.FrameworkDescription`.

Because, if you look at the code for `RuntimeInformation.FrameworkDescription`, you'll notice that the string produced for this property will always contain the constant value of `FrameworkName`, which is `".NET"`:

https://github.com/dotnet/runtime/blob/aeecdb8c00e1a673838b22168c264e896d9446e8/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.cs#L14-L34

(And there doesn't seem any other code that manipulates the `s_frameworkDescription` field aside from the `RuntimeInformation.FrameworkDescription` getter.)

**Reply by normj** [reply]
Sorry I should have been clearer. The `EXPLICIT/ACB7CD69` would be coming from the version part. We strip out the prefix `.NET`. So I guess it is a question why would `versionString = typeof(object).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;` return back something like `EXPLICIT/ACB7CD69`

**Reply by jkotas** [reply, marked as answer]
People can build their own custom version of `System.Runtime.InteropServices.RuntimeInformation.dll` and use it in their app instead of the official one.

**Reply by normj** [reply]
Thanks @jkotas. I think that is what we are seeing. We have seen some Unity crash dumps online that also show `explicit` in the version. That could be what we are seeing.
